engine/src/eth/rpc.rs

```diff
@@ -260,17 +264,29 @@ impl EthSigningRpcApi for EthRpcSigningClient {
 		self.signer.address()
 	}
 
-	async fn send_transaction(&self, mut tx: Eip1559TransactionRequest) -> Result<TxHash> {
+	async fn sign_transaction(&self, mut tx: Eip1559TransactionRequest) -> Result<Bytes> {
 		tx.nonce = Some(self.get_next_nonce().await?);
+		let mut tx = TypedTransaction::Eip1559(tx);
 
-		let res = self.signer.send_transaction(tx, None).await;
-		if res.is_err() {
-			// Reset the nonce just in case (it will be re-requested during next broadcast)
-			tracing::warn!("Resetting eth broadcaster nonce due to error");
-			*self.nonce_info.lock().await = None;
-		}
+		self.signer.fill_transaction(&mut tx, None).await.map_err(anyhow::Error::msg)?;
 
-		Ok(res?.tx_hash())
+		self.signer
+			.signer()
+			.sign_transaction(&tx)
+			.await
+			.map_err(anyhow::Error::msg)
+			.map(|signature| tx.rlp_signed(&signature))
+	}
+
+	async fn send_raw_transaction(&self, tx: Bytes) -> Result<TxHash> {
+		match self.signer.inner().send_raw_transaction(tx).await {
+			Ok(pending_tx) => Ok(pending_tx.tx_hash()),
+			Err(e) => {
+				// Reset the nonce just in case (it will be re-requested during next broadcast)
+				*self.nonce_info.lock().await = None;
```

Comment:
Hmm I think this may be the culprit of the `replacement transaction underpriced`. The nonce is already signed over by this point, so it's strange to try resetting it at the broadcast level, resetting the nonce if there's a signing error would make more sense to me @albert-llimos

Reply:
While I was about to agree with your statment, I actually don't think so after putting in some thought.
Signing won't fail even if the nonce is already used, it will be the broadcast that will fail in that case. Ideally we should only reset the nonce if the transaction broadcast fails due to the nonce having been used before(`replacement transaction underpriced`) and that is from a previous transaction that is still in the mempool. In the case of being the second broadcast retrial of the same tx, then it's intended. Given that we can't really know which scenario it is, it feels like the safest and correct behavior is to just reset the nonce for the next signing.

Reply:
Also, there is another error, something along the lines of `nonce already used` that might be the result of the broadcast step that should also trigger a nonce reset. So the current approach makes sense to me even if it feels very counterintuitive.

Reply:
Any ideas why we would see the `replacement transaction underpriced` in that case? doesn't happen consistently but it has

Reply:
For background , that happens when another transaction is already in the mempool with that nonce. So it's not that we are reusing a nonce used in a mined transaction.
I can't tell the exact reason of why that would happen, I would expect that to be some edge case where we query the latest nonce, which includes pending tx's, and for some reason a pending transaction is not included. However, as long as the error is not `nonce already used` I think it's alright. That would mean that we are using a nonce from a mined transaction.

Reply:
@kylezs @j4m1ef0rd do we have any engine log that has a `replacement transaction underpriced message`? It would be useful to know if a previous transaction has failed to broadcast or what exactly is the behavior around it.

Reply:
https://github.com/chainflip-io/chainflip-backend/actions/runs/8244260007/job/22546404358

Reply:
I just re-read what you had above Albert. You're right re the nonce error only occurring on broadcast, so doesn't make sense to move it. The direction I was thinking though was, because we signed and broadcasted in an atomic step before, it meant that if you signed and it failed to broadcast you'd reset the nonce by signing again. However, now, because we don't reset it on broadcast, it'll just broadcast the same thing again and fail again.

Reply:
> now, because we don't reset it on broadcast, it'll just broadcast the same thing again and fail again.
Yes. That makes sense if the purpose of retrying the broadcast is to handle cases where there is a temporary disconnection or some RPC issue. If it's a nonce issue that doesn't matter indeed.
So as you say, resigning if there is a nonce related error makes sense to me. We need to ensure we cover all the error cases where it can be nonce related. Do you think we can rely on that?
This would be particularly important in 1-node networks to prevent broadcast aborted when there is some nonce issue but it should be very irrelevant in live networks as some other node will be able to broadcast it anyway. We still have to implement the right approach ofc.

Besides that, I struggle to imagine what scenario could cause those `replacement transaction underpriced message`. I have added some debug prints and tried reproducing it locally and I can't reproduce it so far.

Reply:
>  We need to ensure we cover all the error cases where it can be nonce related. Do you think we can rely on that?

Yeah we couldn't really ensure this I don't think, nodes could change their response at any time. 

Question then is - is it better to resign every time (as in before this PR) or do as is currently the case in this PR?

Reply:
Yeah... bit of an annoying one. Really don't wanna introduce flakiness in localnets/CI, but can see why we wouldn't want to sign twice. There ought to be a reason we submitted with the wrong nonce anyway - so probably solving the fact that we didn't update the nonce correctly, but signing once, is the solution

Reply:
I think I agree with that. I will look more into it and see if I can figure out the reason for that.

Reply:
Have you had a chance to look into it @albert-llimos ?

Reply:
Is this just a the problem due to signing and broadcasting being scheduled separartely in the retrier? (I haven't looked how that works in detail, but that seems plausible @kylezs?) It is a little unfortunate that signing a tx has be done as a "request" even thought it is done completely locally...

Reply:
@kylezs I did try many times but I haven't been able to reproduce it, not even one single time. Looking at the code I can't really figure out why that would happen.
I've just been able to come up with one potential scenario that could explain this issue, related to what Maxim mentioned. There is a nonce timeout that forces a re-fetching of the nonce, which effectively looks at transcations mined and in the mempool. 
1. Transaction A is signed with nonce 5
2. Transaction A is requested for broadcast but not broadcasted yet
3. Transaction B is tried to be signed. It hits the timeout and fetches the latest nonce. Since Transaction A is not broadcasted yet, it gets nonce 5
4. Transaction B is now signed with nonce 5.
5. Either Transaction A or B will be rejected by the blockchain client with `replacement transaction underpriced`.

Another possibility, is that a broadcast request has failed, and we therefore reset the nonce, but for some reason the blockchain client has gotten the transaction anyway. Then we are trying to broadcast it again, which will return the mentioned error and I suspect it might end up with a BroadcastAborted.

You guys tell me if any of these two explanations make any sense to you.

Reply:
> It is a little unfortunate that signing a tx has be done as a "request" even thought it is done completely locally...

I guess it doesn't have to be this way. We could put a synchronous signer alongside the retriers for these local requests if we wanted. We can do this. Makes it conceptually clearer for sure.

Reply:
> Is this just a the problem due to signing and broadcasting being scheduled separately in the retrier?

It could be, especially given that when they were scheduled together we didn't see this issue. But I think I'm willing to say this is exceedingly rare. And get this PR in (after improving the above, having the signer be available for signing without going through the retrier (which is intended to be for RPC requests primarily.

Reply:
Yeah they seem reasonable @albert-llimos, as above comment, I think they are quite unlikely and that only signing once is a slight improvement, as we will tend to use less gas on failed txs.